subsys/bluetooth/host/keys.c

```diff
@@ -51,11 +55,28 @@ struct bt_keys *bt_keys_get_addr(u8_t id, const bt_addr_le_t *addr)
 			first_free_slot = i;
 		}
 	}
-
+#if defined (CONFIG_BT_KEY_DISPLACEMENT)
+	if (first_free_slot == ARRAY_SIZE(key_pool)) {
+		size_t oldest_slot = 0;
+		for (i = 1; i < ARRAY_SIZE(key_pool); i++) {
+			if (key_pool[oldest_slot].aging_counter > key_pool[i].aging_counter) {
+				oldest_slot = i;
+			}
+		}
+		bt_unpair(key_pool[oldest_slot].id, &key_pool[oldest_slot].addr);
+		if (!bt_addr_le_cmp(&key_pool[oldest_slot].addr, BT_ADDR_LE_ANY)) {
+			first_free_slot = oldest_slot;
+		}
+	}
+#endif  /* CONFIG_BT_KEY_DISPLACEMENT */
 	if (first_free_slot < ARRAY_SIZE(key_pool)) {
 		keys = &key_pool[first_free_slot];
 		keys->id = id;
 		bt_addr_le_copy(&keys->addr, addr);
+#if defined (CONFIG_BT_KEY_DISPLACEMENT)
+		++aging_counter_val;
+		keys->aging_counter = aging_counter_val;
```

Comment:
I think it'd be equally readable to just do `keys->aging_counter = ++aging_counter_val;`